js/ui/animation.js

```diff
@@ -23,12 +23,12 @@ var Animation = class {
             this._loadFile.bind(this, file, width, height));
 
         this._speed = speed;
-        this._frameTimeouts = [this._speed];
 
         this._isLoaded = false;
         this._isPlaying = false;
         this._timeoutId = 0;
-        this._frame = 0;
+        this._framesInfo = [];
```

Comment:
Please squash this commit with "animation: Add support for different frame timeouts"